### PR TITLE
Add Claims and PropertyBag properties to TokenValidationResult

### LINKED ISSUE #1194: Add property bag to TokenValidationResult
In the new world, it seems `ClaimsIdentity` is no longer king, as `JsonWebTokenHandler` exclusively uses `SecurityTokenDescriptor.Claims` (an "untyped" dictionary) instead of `SecurityTokenDescriptor.Subject`.

Would it make sense to have a property in `TokenValidationResult` exposing an equivalent to `SecurityTokenDescriptor.Claims`?

### PR DESCRIPTION
**Pull request by RojaEnnam**
Fixes #1194

- Adds `Claims `property, that is equivalent to `SecurityTokenDescriptor.Claims` on `TokenValidationResult`.
- Adds `PropertBag`

### CODE REVIEW

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs [hunk 1]
```diff
@@ -65,50 +63,6 @@ public class JwtTokenUtilities
             JwtHeaderParameterNames.Zip
         };
 
-        /// <summary>
-        /// Creates a dictionary from a list of Claim's.
-        /// </summary>
-        /// <param name="claims"> A list of claims.</param>
-        /// <returns> A Dictionary representing claims.</returns>
-        internal static IDictionary<string, object> CreateDictionaryFromClaims(IEnumerable<Claim> claims)
```
Comment (GeoK) [inline src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs:73]: What is the reason for moving this method to M.IM.Tokens?

Comment (RojaEnnam) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs:73]: `TokenValidationResult  ` is in M.IM.Tokens, and to use JwtTokenUtilities it has to refer to M.IM.JsonWebTokens . Which causes circular dependency.

File: src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Constants for Json Web tokens.
+    /// </summary>
+    public static class JsonClaimValueTypes
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs]: This type is in M.IM.JsonWebTokens, we shouldn't have two public types with the same name.

File: src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs [hunk 3]
```diff
@@ -39,6 +41,79 @@ namespace Microsoft.IdentityModel.Tokens
...
+                    payload[jsonClaimType] = jsonClaimValue;
+                }
+            }
+
+            return payload;
+        }
+
+        internal static object GetClaimValueUsingValueType(Claim claim)
+        {
+            if (claim.ValueType == ClaimValueTypes.Boolean && bool.TryParse(claim.Value, out bool boolValue))
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs:112]: most likely, the ValueType will be ClaimValueTypes.String, perhaps we should check that first.

File: src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs (hunk 2 above)
Comment (RojaEnnam) [reply src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs]: JsonClaimValueTypes is used by GetClaimValueUsingValueType , due to circular dependency problem I had to create a new one as System.IdentityModel.Tokens.Jwt also has one :)

File: src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs (hunk 3 above)
Comment (RojaEnnam) [reply src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs:112]: I think that's a good idea if its most likely string. I made the change.

File: src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs (hunk 2 above)
Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs]: I recognize the problem, but I agree with Brent here. I'm not sure this is how we want to solve this.

Comment (RojaEnnam) [reply src/Microsoft.IdentityModel.Tokens/JsonClaimValueTypes.cs]: Defining `JsonClaimValueTypes `consts locally in `TokenUtilities `to solve the problem at this moment.

File: test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationResultTests.cs [hunk 4]
```diff
@@ -45,8 +45,8 @@ public void GetSets()
             TokenValidationResult tokenValidationResult = new TokenValidationResult();
             Type type = typeof(TokenValidationResult);
             PropertyInfo[] properties = type.GetProperties();
-            if (properties.Length != 7)
-                Assert.True(false, "Number of public fields has changed from 7 to: " + properties.Length + ", adjust tests");
+            if (properties.Length != 9)
```
Comment (keegan-caruso) [inline test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationResultTests.cs:48]: +2 properties but +1 test case. Should Claims not be included here?

Comment (RojaEnnam) [reply test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationResultTests.cs:48]: I believe every time a new object is created when accessing Claims, [here ](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/92580d7f1a3027c6f400cc11be46d9e061dedea2/test/Microsoft.IdentityModel.TestUtils/TestUtilities.cs#L161) they don't match on all properties as they are different objects.

**Review by GeoK — APPROVED** [review]

**Review by keegan-caruso — APPROVED** [review]

**Pull request merged**